[TheodoreKooo] iP

TheodoreKooo opened the pull request:
### RizBot

> "An idle mind is the devil's workshop"

### RizBot aims to make the your todo-list more interactive and specific! The to-do bot is,

- Command based, e.g. "todo" for a task with no deadline, "deadline" to add a task with a specific deadline.
- Simple to use
- Funnily sarcastic with its responses!

### To try out the RizBot,

1. Download it from [here](https://github.com/TheodoreKooo/ip/releases/tag/A-Jar) (Download the Riz.Jar file).
2. Launch Terminal and navigate to the folder which it is stored in and run _java -jar Riz.jar_.
3. Add and manage your tasks based off the instructions in the **_README_**.
4. Remember to mark your tasks as done after you have completed! 😃

### Features

- [x] Adding/Deleting tasks(Index-based).
- [x] Support for tasks with deadlines or time windows. 
- [x] Searching for tasks.
- [ ] Usable GUI (coming soon).
- [ ] Deleteing tasks(Name-based).
- [ ] Editing Tasks (coming soon).

Here is the Main class in which all the magic begins!
The main method is 
`public static void main(String[] args) `
```java
public class Riz {
    private TaskList taskList;
    private Storage storage;

    public Riz(String filePath) {
        this.storage = new Storage(filePath);
        this.taskList = new TaskList(this.storage.loadFromFile());
    }

    public void run() {
        Scanner scanner = new Scanner(System.in);
        String dotted = "-----------------------------------";
        //greetings
        Ui.printGreetings();
        boolean running = true;

        while (running) {
            String input = scanner.nextLine();
            if (input.equals("bye")) {
                running = false;
                Ui.printGoodbye();
            } else {
                Parser.parse(scanner, this.taskList, this.storage, input);
            }
        }
    }
   public static void main(String[] args) {
        Riz riz = new Riz("riz/data/riz.txt");
        riz.run();
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

TheodoreKooo committed:
Level 0: Rename, Greet, Exit

TheodoreKooo committed:
Level-1: Echo

TheodoreKooo committed:
Level-2: Add, List

TheodoreKooo committed:
Level-3 : Mark as Done

TheodoreKooo committed:
Try to debug the subclasses of task

Theodorekoo committed:
Level-4: ToDo, Event, Deadline

TheodoreKooo committed:
Level-5 Handling the basic exceptions

TheodoreKooo committed:
Level-5 all possible exceptions handled

TheodoreKooo committed:
completed the deletion function

TheodoreKooo committed:
no message

Theodorekoo committed:
removed the dotted lines between commands to make it more aesthetically pleasing

Theodorekoo committed:
Working on the writeToFile and ReadToFile methods

Theodorekoo committed:
Completed the save function for Level 7, still left with the case where the data file does not exist.

Theodorekoo committed:
Level-7: Save completed

Theodorekoo committed:
Merge branch 'branch-Level-7'

Theodorekoo committed:
Level-8: Dates and Times

Theodorekoo committed:
finished up level 8 and added a new feature "clear"

Theodorekoo committed:
Merge branch 'branch-Level-8'

Theodorekoo committed:
finished the extraction of classes Ui, Storage, Parser and TaskList

Theodorekoo committed:
Merge branch 'MoreOOP'

Theodorekoo committed:
made all of the subclasses ToDo, Deadline and Event public

Theodorekoo committed:
Completed organizing them into packages, they are catagorized by the 2 packages "riz.data" and "riz.io" as of now.
The Classes Deadline, Event, Storage, Task, TaskList and Todo are in the package "riz.data".
The classes Parser, RizException and Ui are in the package"riz.io".

Theodorekoo committed:
Merge branch 'Packages'

Theodorekoo committed:
Merge branch 'add-gradle-support'

# Conflicts:
#	docs/README.md

TheodoreKooo committed:
Created Test files to test the methods in the various classes

TheodoreKooo committed:
added in the tests for the Parser class

Theodorekoo committed:
Pakaged the app as a JAR file

Theodorekoo committed:
Merge branch 'A-Jar'

TheodoreKooo committed:
no message

TheodoreKooo committed:
Implemented Level-9: Find

TheodoreKooo committed:
Added JavaDocs

taeewonnn commented on src/main/java/riz/data/Deadline.java:
```diff
@@ -0,0 +1,19 @@
+    private LocalDateTime by;
+    public Deadline (String deadline, String by) {
+        super(deadline);
+        DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
```
Should the method variable be in verb form? 🤔

taeewonnn commented on src/main/java/riz/data/Task.java:
```diff
@@ -0,0 +1,33 @@
+
+public abstract class Task {
+    private String task;
+    private boolean completed;
```
Perhaps, the boolean variable name can be more specific. For eg "isComplete".

TheodoreKooo replied:
Yup thanks for the headsup!

taeewonnn commented on src/main/java/riz/io/Parser.java:
```diff
@@ -0,0 +1,128 @@
+            if (!isNumber) {
+                Ui.markError();
+            }
+            int curr = Integer.parseInt(token[1]) - 1;
```
Should the name of this integer be more specific?

taeewonnn commented on src/main/java/riz/io/Parser.java:
```diff
@@ -0,0 +1,128 @@
+                    System.out.println("added: " + task + "...");
+                }
+            } else {
+                Ui.yapError();
```
Perhaps the naming of this yapError method avoid "texting style".

TheodoreKooo replied:
Thanks for all of the above suggestions! I will improve on the code based on the feedback.

taeewonnn commented on src/main/java/riz/io/Ui.java:
```diff
@@ -0,0 +1,59 @@
+        System.out.println("Bye... Hope to see you again...\n");
+    }
+
+    public static void printClearConfirmation() {
```
I like how the names of the methods here are very specific

taeewonnn reviewed (commented):
Overall, very creative coding names and good code quality.

Theodorekoo committed:
no message

Theodorekoo committed:
Merge branch 'master' of https://github.com/TheodoreKooo/ip

Theodorekoo committed:
Merge remote-tracking branch 'origin/Level-9-Find'

Theodorekoo committed:
Implemented the base GUI, made to echo any commands. To do next: make sure the appropriate responses are given based on the instructions to the bot

Theodorekoo committed:
no message

Theodorekoo committed:
Managed to make the GUI respond in the same way as on the console by returning strings instead of just printing to the console

Theodorekoo committed:
Added MainWindow.fxml and DialogBox.fxml

Theodorekoo committed:
no message

alex-setyawan commented on src/main/java/riz/data/Event.java:
```diff
@@ -0,0 +1,32 @@
+    @Override
+    public String toString() {
+        DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("dd MMM yyyy hh:mm a");
+        return "E" + super.toString() + " | " + this.from.format(outputFormatter) + " - " + this.to.format(outputFormatter);
```
Perhaps you would like to have multiple line breaks (like before each +) to better adhere to the Java coding standard?

Theodorekoo committed:
Completed basic implementation of GUI

Theodorekoo committed:
Made the first button for the "todo" task. This is to make the GUI easier to use

Theodorekoo committed:
fixed small bug relating to todo button

Theodorekoo committed:
Completed the implementations of the other buttons

Theodorekoo committed:
added tags

Theodorekoo committed:
Merge branch 'Level-10-GUI'

Theodorekoo committed:
Fixed small bug on the clearList button

TheodoreKooo committed:
Merge branch 'master' of https://github.com/TheodoreKooo/ip

Theodorekoo committed:
Added assertions to various portions of the code.

Classes in which assertions were added to
* Task
* ToDo
* Deadline
* Event
* Parser
* Storage

TheodoreKooo committed:
Merge pull request #2 from TheodoreKooo/branch-A-Assertions

Added assertions to various portions of the code.

Theodorekoo committed:
Improving on the coding quality

The Parser class has a method that is too long and has too deep
nesting, making it difficult to read or modify code in the future.

Implemented the same logic with a switch statement instead.
Created separate methods for each case.

This is so that the code is structured in such a way that is easier
to read, organize and modify in the future.

Theodorekoo committed:
Merge branch 'branch-A-Assertions' of https://github.com/TheodoreKooo/ip into branch-A-Assertions

Theodorekoo committed:
Merge branch 'branch-A-Assertions'

Theodorekoo committed:
The code in the storage class is too deeply nested and convoluted.

It has to be changed in case that more types of tasks are added,
then we can add them easier.

The three different tasks that can be created has been abstracted
out into a method.

It is easier to segregate and organize the code and where to focus
on should we decide to look into or modify the code in the future.

Theodorekoo committed:
no message

Theodorekoo committed:
Merge branch 'branch-A-CodeQuality'

Theodorekoo committed:
Extension C-Help done

Theodorekoo committed:
Took a screenshot of the GUI and uploaded into docs as UI.png

TheodoreKooo committed:
Worked on the readme

TheodoreKooo committed:
Completed the userguide

Theodorekoo committed:
Did some refinements in the code and removed some classes and
methods that had no use cases.
Also updated the README to include instrucions on how to use the
JAR File.

Theodorekoo committed:
no message

Theodorekoo committed:
no message

Theodorekoo committed:
no message

Theodorekoo committed:
changed the readme

Theodorekoo committed:
changed the readme again to make sure that the number points are
correct

Theodorekoo committed:
Reduced the height of the GUI.
Changed the JUnit test as there was a small bug in the tests.

Theodorekoo committed:
Refactored the Exception Handling into neater chunks of code.

Theodorekoo committed:
Made more updates to the exception checking.

Theodorekoo committed:
Dealt with various edge cases of the exception handling.

Theodorekoo committed:
Removed the dead code for the Button fields in MainWindow.java

Theodorekoo committed:
Added in the "bye" command.

Theodorekoo committed:
Changed the code to account for an potential fatal error for the
todo task.

Theodorekoo committed:
Removed even more dead code from the import statements.

TheodoreKooo committed:
Update README.md

TheodoreKooo committed:
Update README.md

TheodoreKooo committed:
Update README.md
